ui/wobblyFx.js

```diff
@@ -0,0 +1,609 @@
+/* exported enable, disable */
+'use strict';
+
+/**
+ * This code is copied from:
+ * https://github.com/hermes83/compiz-alike-windows-effect
+ *
+ */
+
+const { GLib, GObject, Clutter, Meta } = imports.gi;
+const ExtensionUtils = imports.misc.extensionUtils;
+const Extension = ExtensionUtils.getCurrentExtension();
+const Hack = ExtensionUtils.getCurrentExtension();
+const Settings = Hack.imports.utils.getSettings();
+
+const CLUTTER_TIMELINE_DURATION = 1000 * 1000;
+const CORNER_RESIZING_DIVIDER = 6;
+
+function Prefs() {
+    this.FRICTION = {
+        key: 'wobbly-spring-friction',
+        get: function () { return Settings.get_double(this.key); },
+        set: function (v) { Settings.set_double(this.key, v); },
+        changed: function (cb) { return Settings.connect('changed::' + this.key, cb); },
+        disconnect: function () { return Settings.disconnect.apply(settings, arguments); },
+    };
+
+    this.SPRING = {
+        key: 'wobbly-spring-k',
+        get: function () { return Settings.get_double(this.key); },
+        set: function (v) { Settings.set_double(this.key, v); },
+        changed: function (cb) { return Settings.connect('changed::' + this.key, cb); },
+        disconnect: function () { return Settings.disconnect.apply(settings, arguments); },
+    };
+
+    this.MANUAL_RESTORE_FACTOR = {
+        key: 'wobbly-slowdown-factor',
+        get: function () { return Settings.get_double(this.key); },
+        set: function (v) { Settings.set_double(this.key, v); },
+        changed: function (cb) { return Settings.connect('changed::' + this.key, cb); },
+        disconnect: function () { return Settings.disconnect.apply(settings, arguments); },
+    };
+
+    this.SKIP_FRAMES_BEFORE_SPRING_START = {
+        key: 'skip-frames-before-spring-start',
+        get: function () { return 1; },
+        set: function () {},
+        changed: function () {},
+        disconnect: function () {},
+    };
+
+    this.MAXIMIZE_EFFECT_ENABLED = {
+        key: 'maximize-effect-enabled',
+        get: function () { return true; },
+        set: function () {},
+        changed: function () {},
+        disconnect: function () {},
+    };
+
+    this.RESIZE_EFFECT_ENABLED = {
+        key: 'resize-effect-enabled',
+        get: function () { return true; },
+        set: function () {},
+        changed: function () {},
+        disconnect: function () {},
+    };
+
+    this.X_TILES = {
+        key: 'x-tiles',
+        get: function () { return 6; },
+        set: function () {},
+        changed: function () {},
+        disconnect: function () {},
+    };
+
+    this.Y_TILES = {
+        key: 'y-tiles',
+        get: function () { return 4; },
+        set: function () {},
+        changed: function () {},
+        disconnect: function () {},
+    };
+};
```

Comment:
Since we're already modifying the extension, should we also remove Prefs?

Reply:
I've removed the Prefs object.